Classes/Event/EventMetadata.php

```diff
@@ -1,112 +0,0 @@
-<?php
-namespace Neos\Cqrs\Event;
-
-/*
- * This file is part of the Neos.Cqrs package.
- *
- * (c) Contributors of the Neos Project - www.neos.io
- *
- * This package is Open Source Software. For the full copyright and license
- * information, please view the LICENSE file which was distributed with this
- * source code.
- */
-
-/**
- * The EventMetadata is a container for arbitrary metadata for Events.
- * This class is immutable.
- *
- * @api
- */
-class EventMetadata
```

Comment:
I fear we are talking cross purpose here.. I didn't mean "thingy" in a snarky way.
I agree that the event that you process should be immutable, and it *is*. But before we store the event we want to be able to *change* it's metadata. This just can't work with immutability by definition. And using immutable objects that you then replace has just the same semantics. Especially now that the metadata is only ever accessible via one (low-level) extension point I don't understand your actual concerns